apps/workers/bigquery/worker.go

```diff
@@ -0,0 +1,279 @@
+		}
+		b, _ := io.ReadAll(f)
+		query := string(b)
+		if isGCSExport(w.DestPath) {
+			query = addExportToGCS(b, w.DestPath, bigquery.CSV)
+		} else if w.DestPath != "" {
+			w.writeToFile = true
+		}
+		return w.Query(ctx, client, query)
+	case ".json":
+		format = bigquery.JSON
+		return w.Load(ctx, client, format)
+	case ".csv":
+		format = bigquery.CSV
+		return w.Load(ctx, client, format)
+	}
+
+	return task.Failf("unsupported file format %v, expected:sql|json|csv", filepath.Ext(w.File))
+}
+
+// isGCSExport checks the url path to see it is meant to be exported to GCS through BigQuery
+// - starts with gs://
+// - must contain a single asterisk * in the filename
+func isGCSExport(s string) bool {
+	if !strings.HasPrefix(s, "gs://") {
+		return false
+	}
+	d, f := filepath.Split(s)
+	return !strings.Contains(d, "*") && strings.Count(f, "*") == 1
+}
+
+func (w *worker) Load(ctx context.Context, client *bigquery.Client, format bigquery.DataFormat) (task.Result, string) {
+	var loader *bigquery.Loader
+	if w.FromGCS { // load from Google Cloud Storage
+		gcsRef := bigquery.NewGCSReference(w.File)
+		gcsRef.SourceFormat = format
+		loader = w.DestTable.BqTable(client).LoaderFrom(gcsRef)
+	} else { // load from file reader
+		r, err := file.NewReader(w.File, &w.Fopts)
+		if err != nil {
+			return task.Failf("problem with file: %s", err)
+		}
+		bqRef := bigquery.NewReaderSource(r)
+		bqRef.SourceFormat = format
+		bqRef.MaxBadRecords = 1
+		loader = w.DestTable.BqTable(client).LoaderFrom(bqRef)
+	}
+
+	loader.WriteDisposition = bigquery.WriteAppend
+	if len(w.DeleteMap) > 0 {
+		q := delStatement(w.DeleteMap, w.DestTable)
+		j, err := client.Query(q).Run(ctx)
+		if err != nil {
+			return task.Failf("delete statement: %s", err)
+		}
+		status, err := j.Wait(ctx)
+		if err != nil {
+			return task.Failf("delete wait: %s", err)
+		}
+		if status.Err() != nil {
+			return task.Failf("delete: %s", err)
+		}
+		status = j.LastStatus()
+		if qSts, ok := status.Statistics.Details.(*bigquery.QueryStatistics); ok {
+			w.SetMeta("rows_del", strconv.FormatInt(qSts.NumDMLAffectedRows, 10))
+		}
+	}
+
+	if w.Truncate {
+		loader.WriteDisposition = bigquery.WriteTruncate
+	}
+
+	job, err := loader.Run(ctx)
+	if err != nil {
+		return task.Failf("loader run: %s", err)
+	}
+	status, err := job.Wait(ctx)
+	if err == nil {
+		if status.Err() != nil {
+			return task.Failf("job completed with error: %v", status.Errors)
+		}
+		if sts, ok := status.Statistics.Details.(*bigquery.LoadStatistics); ok {
+			w.SetMeta("rows_insert", strconv.FormatInt(sts.OutputRows, 10))
+			return task.Completed("%d rows (%s) loaded", sts.OutputRows, humanize.Bytes(uint64(sts.OutputBytes)))
+		}
+	}
+
+	return task.Completed("completed")
+}
+
+// delStatement used to delete data before insertion into table (see delete option)
+func delStatement(m map[string]string, d Destination) string {
+	s := make([]string, 0)
+	for k, v := range m {
+		s = append(s, k+" = "+v)
+	}
```

Comment:
do we need to worry about any sql injection here? Maybe use parameterizing queries here?